### PR TITLE
+ht2 #758 allow early priotity frames

### PR DESCRIPTION
**Pull request by ktoso**
Refs #758, only the allowing part. 
Also added debug information to all GOAWAYs

### CODE REVIEW

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala [hunk 1]
```diff
@@ -152,12 +151,16 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
 
               dispatchSubstream(Http2SubStream(frame, data))
 
+            case PriorityFrame(streamId, exclusiveFlag, streamDependency, weight) ⇒
+              // must be before the "unknown stream id" case, since can be sent eagerly (e.g. firefox does so)
+              debug(s"Received PriorityFrame for stream $streamId with ${if (exclusiveFlag) "exclusive " else "non-exclusive "} dependency on stream $streamDependency and weight $weight")
```
Comment (ktoso) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: this move is actually enough to keep firefox happy

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala [hunk 2]
```diff
@@ -152,12 +151,16 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
...
+
             case e: StreamFrameEvent if !incomingStreams.contains(e.streamId) ⇒
               // if a stream is invalid we will GO_AWAY
-              pushGOAWAY()
+              pushGOAWAY(ErrorCode.PROTOCOL_ERROR, "Unknown stream id: " + e.streamId)
 
             case h: ParsedHeadersFrame ⇒
-              pushGOAWAY()
+              pushGOAWAY(ErrorCode.PROTOCOL_ERROR, "Unexpected internal frame reached Demux! Was: " + h)
```
Comment (jrudolph) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: The error message here is not quite correct. When we receive it here it is probably a trailing header frame here which we currently don't support but which we should ignore. It's ok to leave it like this but add a TODO comment to fix it in the course of #531.

File: akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala [hunk 3]
```diff
@@ -573,7 +573,7 @@ class Http2ServerSpec extends AkkaSpec("" + "akka.loglevel = debug")
         sendSETTING(SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, 1)
         expectSettingsAck()
 
-        // TODO actually apply the limiting and verify it works
+        // TODO actually apply the limiting and verify it works
```
Comment (jrudolph) [inline akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala]: extra whitespace here

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala (hunk 1 above)
Comment (jrudolph) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: :+1:

File: akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala (hunk 2 above)
Comment (ktoso) [inline akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala]: Will add a todo explaining that 👍

**Comment by jrudolph** [comment]
LGTM (maybe add the small comment)

**Pull request merged**